### PR TITLE
Adding oidc (dex style) authentication to fleet

### PR DESCRIPTION
**Pull request by naemono**
Without this you get

```
$ kubectl fleet details mycluster
API server endpoint: https://k8s-master.mycluster.me.cloud:443
No Auth Provider found for name "oidc"
```

### CODE REVIEW

**Comment by mhausenblas** [comment]
@naemono thanks a lot for your contrib! Out of interest, in which environment and with which Kubernetes distro does this happen?

File: cmd/fleet/main.go [hunk 1]
```diff
@@ -2,7 +2,8 @@ package main
 
 import (
 	"github.com/mhausenblas/kcf/cmd/fleet/cli"
-	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp" // required for GKE
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"  // required for GKE
```
Comment (corneliusweig) [inline cmd/fleet/main.go]: Isn't this enough for all auth providers?
```go
_ "k8s.io/client-go/plugin/pkg/client/auth"
```

Comment (mhausenblas) [reply cmd/fleet/main.go]: Yeah, good point! @naemono do you want to test this?

**Comment by naemono** [comment]
My apologies.  I totally missed the updates to this.  Will test

**Comment by naemono** [comment]
@mhausenblas After testing, the single import works.  Also, these are custom built kubernetes clusters with in-house tools, integrated with https://github.com/dexidp/dex

**Comment by mhausenblas** [comment]
Perfect, thank you for this @naemono!

**Pull request merged**